scripts/host-check

```diff
@@ -1169,6 +1181,68 @@ def check_shmem_pages_max_size() -> CheckFuncReturn:
     return CheckState.SUCCESS, f"{shared_mem_max_page_size:.1f} GiB"
 
 
+def check_module_non_default_params() -> CheckFuncReturn:
+    """
+    Checks the kernel modules have been loaded with default runtime parameters
+    (Checks parameters in MODULE_DEFAULT_PARAM)
+    """
+    all_modules_str = ""
+    for module, param_defaults in MODULE_DEFAULT_PARAM.items():
+        module_str = ""
+        if not _is_module_loaded(
+            module.replace("-", "_")
+        ) and not _is_module_builtin(module):
+            # Skip module param check if module is not loaded
+            continue
+        # sysfs filesystem requires the underscore version of the kernel module
+        # name
+        module = module.replace("-", "_")
+        all_params_str, _ = run_cmd(["modinfo", "--field", "parm", module])
+        param_non_default = []
+        param_check_fail = []
+        for param, def_val in param_defaults.items():
+            if not re.search(rf"^{param}:", all_params_str, re.MULTILINE):
+                # if 'param' is absent in modinfo output, it means either the
+                # parameter was introduced in a later kernel version, or the
+                # specific linux distribution does not support the parameter.
+                continue
+            try:
+                path = f"/sys/module/{module}/parameters/{param}"
+                with open(path, "r", encoding="utf-8") as f:
+                    param_val = f.read().strip()
+                # (null) implies the parameter isn't set, and is thus the
```

Comment:
The `intr_mode` param of `igb_uio` can be `(null)`, and that is set to the default value `msix`.

I could change `MODULE_DEFAULT_PARAM` to map to a set of default values instead of a simple string, but this felt cleaner. Can check if that'll be better?

Also, should I document this in the `host-check` script? As in mention the specific parameter `intr_mode`?